tests/base/test_components.py

```diff
@@ -179,6 +180,140 @@ def test_init_ComponentError_given_duplicated_component_name(self):
+    def test_filter_components_does_filter_on_single(self):
+        parent = Component("Parent")
+        child1 = Component("Child1", parent=parent)
+        child2 = Component("Child2", parent=parent)
+
+        child1a = Component("child1A", parent=child1)
+        child1b = Component("child1B", parent=child1)
+
+        def attach_dims_and_physical_comps_to(comp: Component):
+            xy = Component("xy", parent=comp)
+            xz = Component("xz", parent=comp)
+            xyz = Component("xyz", parent=comp)
+
+            PhysicalComponent(
+                "pc_xy",
+                parent=xy,
+                shape="pc_xy shape",
+                material="pc_xy material",
+            )
+            PhysicalComponent(
+                "pc_xz",
+                parent=xz,
+                shape="pc_xz shape",
+                material="pc_xz material",
+            )
+            PhysicalComponent(
+                "pc_xyz",
+                parent=xyz,
+                shape="pc_xyz shape",
+                material="pc_xyz material",
+            )
+
+        attach_dims_and_physical_comps_to(child1a)
+        attach_dims_and_physical_comps_to(child1b)
+        attach_dims_and_physical_comps_to(child2)
+
+        parent.filter_components(["xz"])
+
+        xy = parent.get_component("xy")
+        xz = parent.get_component("xz")
+        xyz = parent.get_component("xyz")
+
+        assert xy is None
+        assert type(xz) is Component
+        assert xyz is None
+
+    def test_filter_components_does_filter_on_double(self):
+        parent = Component("Parent")
+        child1 = Component("Child1", parent=parent)
+        child2 = Component("Child2", parent=parent)
+
+        child1a = Component("child1A", parent=child1)
+        child1b = Component("child1B", parent=child1)
+
+        def attach_dims_and_physical_comps_to(comp: Component):
+            xy = Component("xy", parent=comp)
+            xz = Component("xz", parent=comp)
+            xyz = Component("xyz", parent=comp)
+
+            PhysicalComponent(
+                "pc_xy",
+                parent=xy,
+                shape="pc_xy shape",
+                material="pc_xy material",
+            )
+            PhysicalComponent(
+                "pc_xz",
+                parent=xz,
+                shape="pc_xz shape",
+                material="pc_xz material",
+            )
+            PhysicalComponent(
+                "pc_xyz",
+                parent=xyz,
+                shape="pc_xyz shape",
+                material="pc_xyz material",
+            )
+
+        attach_dims_and_physical_comps_to(child1a)
+        attach_dims_and_physical_comps_to(child1b)
+        attach_dims_and_physical_comps_to(child2)
+
+        parent.filter_components(["xy", "xz"])
+
+        xy = parent.get_component("xy")
+        xz = parent.get_component("xz")
+        xyz = parent.get_component("xyz")
+
+        assert type(xy) is Component
+        assert type(xz) is Component
```

Comment:
I'm sure you know this but I didnt suggest the `isinstance` change here because it would return `True` for child classes too.